File: frontend/event-ticketing-app/src/api/userManagementService.ts
import api from './axios';

export interface UserDTO {
  id: number;
  name: string;
  email: string;
  phoneNumber?: string;
  address?: string;
  isActive: boolean;
  roles: string[];
  createdAt: string;
}

const userManagementService = {
  // Get all users (Admin only)
  getAllUsers: async (): Promise<UserDTO[]> => {
    const response = await api.get('/users');
    return response.data.data;
  },

  // Promote user to organizer (Admin only)
  promoteToOrganizer: async (userId: number): Promise<UserDTO> => {
    const response = await api.put(`/users/${userId}/promote-organizer`);
    return response.data.data;
  },

  // Demote organizer to user (Admin only)
  demoteFromOrganizer: async (userId: number): Promise<UserDTO> => {
    const response = await api.put(`/users/${userId}/demote-organizer`);
    return response.data.data;
  },

  // Delete user (Admin only)
  deleteUser: async (userId: number): Promise<void> => {
    const response = await api.delete(`/users/${userId}`);
    return response.data;
  },
};

export default userManagementService;

File: backend/eventticketingplatform/src/main/java/com/nipapager/eventticketingplatform/user/service/UserManagementService.java
package com.nipapager.eventticketingplatform.user.service;

import com.nipapager.eventticketingplatform.response.Response;
import com.nipapager.eventticketingplatform.user.dto.UserDTO;

import java.util.List;

/**
 * Service interface for user management operations
 */
public interface UserManagementService {
    Response<List<UserDTO>> getAllUsers();
    Response<UserDTO> promoteToOrganizer(Long userId);
    Response<Void> deleteUser(Long userId);
    Response<UserDTO> demoteFromOrganizer(Long userId);
}